### DISCUSSION TITLE
Создаётся неполная дискография исполнителя

### DISCUSSION THREAD

**Discussion by kisoid** [Chimildic/YaMuTools#54] [Помощь с ошибкой] [answered]
При создании Дискографии в плейлист не попадают треки, входящие только в сборники (и не входящие в альбомы).

Как воспроизвести проблему:
1) заходим на https://music.yandex.ru/artist/904392
2) нажимаем Дискография -> Все треки

попадаем в образовавшийся плейлист и видим, что в него входят треки 5 альбомов и нет многочисленных треков, которые представлены только на сборниках, например этих:
https://music.yandex.ru/album/3831505/track/4771358
https://music.yandex.ru/album/7450576/track/52746183

Раньше всё было нормально, но затрудняюсь сказать, с какой версии началась проблема.

**Comment by kisoid** [comment]
UPD

Нашёл временно для себя в API способ получить _все_ треки вне зависимости от их нахождения/не нахождения в альбомах/сборниках:
https://api.music.yandex.net/artists/904392/tracks?page_size=1000

осталось только прикрутить это в расширение...

**Comment by Chimildic** [comment]
В текущей реализации используется следующее API, из ответа которого берутся только `albums`. 
```
/handlers/artist.jsx?artist=904392&what=albums
```
Насколько вижу есть и `alsoAlbums`, в которых как раз перечисляются сборники. Раньше исходил из того, что сборники содержат либо дубль, либо муть ремиксовую. То что там целый набор уникальных треков странно. Насколько это частое явление? Особенно для популярных исполнителей.

**Reply by kisoid** [reply]
> Раньше исходил из того, что сборники содержат либо дубль, либо муть ремиксовую.

Ну... очевидно, что это не так :)
Опять же, "муть ремиксовая" - это, наверное, _понятие_ характерное для определённых музыкальных мировоззрений, скажем так :) Например, для любителей рока.
А для некоторых меломанов/стилей/направлений именно в этой мути может быть самый цимес.

> То что там целый набор уникальных треков странно. Насколько это частое явление? Особенно для популярных исполнителей.

Насчёт популярных не знаю. Среди моих (chillout & lounge) достаточно ощутимо встречается. Потому что много малоизвестных исполнителей и они по многочисленным сборникам разбросаны.
Есть даже исполнители, все треки которых только в сборниках и представлены на текущий момент.
Например, https://music.yandex.ru/artist/800997/albums

Опять же, редакция ЯМузыки по ошибке легко может поставить галку сборника на полноценном альбоме. Видел такое.

---

Всё-таки понятие "Все треки" должно включать действительно _все_ треки, как мне кажется. По-хорошему, даже все версии всех треков.
Страница https://api.music.yandex.net/artists/904392/tracks?page_size=1000 выдаёт уникальные по id треки, без учёта версий - 99 штук.
То же самое на странице https://music.yandex.ru/artist/904392/tracks - 99 штук
А вот если сделать сортировку по названию трека, то здесь: https://music.yandex.ru/artist/904392/tracks?sort=title&dir=asc
уже 114 треков, т.е. плюсуются разные версии на разных альбомах (которые, кстати, могут отличаться по длительности, например)

**Reply by Chimildic** [reply]
Хорошо, что-нибудь сделаю с этим позднее

**Comment by kisoid** [comment, marked as answer]
Нашёл способ создать плейлист со всеми треками не прибегая к расширению или скриптам, только средствами самой Я.Музыки:

1) заходим на страницу исполнителя, где представлены все треки: https://music.yandex.ru/artist/xxxxxxx/tracks
2) нажимаем воспроизведение первого трека
3) нажимаем кнопку "Очередь прослушивания" (снизу слева, справа от кнопки "Пропустить")
4) убеждаемся, что вверху выбрана вкладка "Очередь прослушивания", а не "История" и написано "Сейчас играет: Популярные треки исполнителя ..."
5) нажимаем "Сохранить как плейлист"
